src/main/java/org/sagebionetworks/bridge/hibernate/HibernateSurveyElement.java

```diff
@@ -0,0 +1,114 @@
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableList;
+
+@Entity
+@Table(name = "SurveyElements")
+public class HibernateSurveyElement implements SurveyElement {
+    @Id
+    private String guid;
+    private String surveyGuid;
+    private String identifier;
+    private String type;
+    private int order;
+    @JsonIgnore
+    private String surveyCompoundKey;
+    
+    @Column(columnDefinition = "mediumtext", name = "data", nullable = true)
+    @Convert(converter = JsonNodeAttributeConverter.class)
+    private JsonNode data;
+    
+    @Column(columnDefinition = "mediumtext", name = "beforeRules", nullable = true)
+    @Convert(converter = SurveyRuleListConverter.class)
+    private List<SurveyRule> beforeRules;
+    
+    @Column(columnDefinition = "mediumtext", name = "afterRules", nullable = true)
+    @Convert(converter = SurveyRuleListConverter.class)
+    private List<SurveyRule> afterRules;
+    
+    /**
+     * No args constructor, required and used by Hibernate for full object initialization.
+     */
+    public HibernateSurveyElement() {}
+    
+    public String getSurveyGuid() {
+        return surveyGuid;
+    }
+    public void setSurveyGuid(String surveyGuid) {
+        this.surveyGuid = surveyGuid;
+    }
+
+    public String getGuid() {
+        return guid;
+    }
+    public void setGuid(String guid) {
+        this.guid = guid;
+    }
+    
+    public String getIdentifier() {
+        return identifier;
+    }
+    public void setIdentifier(String identifier) {
+        this.identifier = identifier;
+    }
+    
+    public int getOrder() {
+        return order;
+    }
+    public void setOrder(int order) {
+        this.order = order;
+    }
+    
+    public String getType() {
+        return type;
+    }
+    public void setType(String type) {
+        this.type = type;
+    }
+    
+    public JsonNode getData() {
+        return data;
+    }
+    public void setData(JsonNode data) {
+        this.data = data;
+    }
+    
+    public List<SurveyRule> getBeforeRules() {
+        return (this.beforeRules == null) ? null : ImmutableList.copyOf(this.beforeRules);
+    }
+    public void setBeforeRules(List<SurveyRule> beforeRules) {
+        this.beforeRules = beforeRules;
+    }
+    
+    public List<SurveyRule> getAfterRules() {
+        return (this.afterRules == null) ? null : ImmutableList.copyOf(this.afterRules);
+    }
+    public void setAfterRules(List<SurveyRule> afterRules) {
+        this.afterRules = afterRules;
+    }
+    
+    public String getSurveyCompoundKey() {
+        return surveyCompoundKey;
+    }
+    
+    public void setSurveyCompoundKey(String surveyCompoundKey) {
+        this.surveyCompoundKey = surveyCompoundKey;
+    }
+    
+    public void setSurveyKeyComponents(String surveyGuid, long createdOn) {
+        this.surveyCompoundKey = surveyGuid + ":" + Long.toString(createdOn);
+    }
```

Comment:
It looks like these are only called by the Dynamo implementations. Would it be cleaner to just delete them from the interface?

Reply:
Removing them would require casting to DynamoSurveyElements everywhere the these methods are called and adding them to SurveyQuestion and SurveyInfoScreen. Should I follow through with the refractor?

Reply:
The alternative would be to make sure get/setSurveyCompoundKey work as intended, which means that we can derive the surveyCompoundKey from the surveyGuid/createdOn and setting the surveyCompoundKey also sets the surveyGuid/createdOn.

Reply:
Can we assume the given surveyCompoundKey when setting it is formatted correctly so that I can split on ":" and set surveyGuid/createdOn?

Reply:
I don't think we can.

Reply:
I'll add a check for that then. Also, this means setting the surveyGuid or createdOn also sets the surveyCompoundKey?

Reply:
Yes, I think it should.

That said, you might consider not storing surveyCompoundKey and just generating it on the fly whenever the getter is called.

Reply:
Should I throw some error if the surveyCompoundKey is formatted incorrectly?